packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -35,6 +35,53 @@ const int _kPubExitCodeUnavailable = 69;
 
 typedef MessageFilter = String? Function(String message);
 
+/// targetPath is the directory in which the content of the extraPath will be moved in
+void joinCaches({
+  required FileSystem fileSystem,
+  required String globalCachePath,
+  required String localCachePath,
+}) {
+  final Directory globalDirectory = fileSystem.directory(globalCachePath);
+  final Directory localDirectory = fileSystem.directory(localCachePath);
+
+  for (final FileSystemEntity entity in localDirectory.listSync()) {
+    final String newPath = fileSystem.path.join(globalDirectory.path, entity.basename);
+    if (entity is File) {
+      final File file = fileSystem.file(entity.path);
+      file.copySync(newPath);
+    } else if (entity is Directory) {
+      final Directory currentDirectory = fileSystem.directory(entity.path);
+      if (!currentDirectory.existsSync()) {
```

Comment:
I don't think this can ever not exist--feel free to explain if i'm reading this wrong.